dipy/core/wavelet.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+from dipy.denoise import nlmeans_block
+
+'''
+ Functions for Wavelet Transforms in 3D domain
+
+ Code adapted from
+
+ WAVELET SOFTWARE AT POLYTECHNIC UNIVERSITY, BROOKLYN, NY
+ http://taco.poly.edu/WaveletSoftware/
+'''
+
+
+def cshift3D(x, m, d):
+    '''
+    3D Circular Shift
+
+    USAGE:
+       y = cshift3D(x, m, d)
+    INPUT:
+       x - N1 by N2 by N3 array
+       m - amount of shift
+       d - dimension of shift (d = 1,2,3)
+    OUTPUT:
+       y - array x will be shifed by m samples down
+           along dimension d
+
```

Comment:
Hi @riddhishb. Please correct docstring to fit the style of our other docstrings.